vsphere-manager-blueprint.yaml

```diff
@@ -0,0 +1,858 @@
+
+
+
+  #############################
+  # Dev Inputs
+  #############################
+
+  # For development purposes, you can use these, which will override the modules
+  # supplied within the rpm.
+  # These should be pip installable tar.gz files.
+  # Note that `install_python_compilers` must not be an empty string if one of the
+  # below modules require compilation.
+  install_python_compilers:
+    type: string
+    default: ''
+
+  cli_source_url:
+    type: string
+    default: ''
+
+  rest_service_source_url:
+    type: string
+    default: ''
+
+  dsl_parser_source_url:
+    type: string
+    default: ''
+
+  rest_client_source_url:
+    type: string
+    default: ''
+
+  plugins_common_source_url:
+    type: string
+    default: ''
+
+  script_plugin_source_url:
+    type: string
+    default: ''
+
+  securest_source_url:
+    type: string
+    default: ''
+
+  amqpinflux_source_url:
+    type: string
+    default: ''
+
+  agent_source_url:
+    type: string
+    default: ''
+
+  #############################
+  # Upload Resources Inputs
+  #############################
+  plugin_resources:
+    description: >
+      Holds any archives that should be uploaded to the manager.
+    default: []
+
+  dsl_resources:
+    description:
+      Holds a set of dsl required resources
+    default: []
+
+######################################################################
+# This is a simple blueprint specific node_type to allow us to define
+# The public_ip of the machine is an accessible property of the host.
+# By default, the `ip` property is the private ip.
+######################################################################
+node_types:
+
+  connection_configuration:
+     derived_from: cloudify.nodes.Root
+     properties:
+       connection_config: {}
+
+######################################################################
+# These nodes comprise the manager's infrastructure and components
+######################################################################
+node_templates:
+  connection_configuration:
+    type: connection_configuration
+    properties:
+      connection_config:
+        username: { get_input: vsphere_username }
+        password: { get_input: vsphere_password }
+        host: { get_input: vsphere_host }
+        port: { get_input: vsphere_port }
+        datacenter_name: { get_input: vsphere_datacenter_name }
+        resource_pool_name: { get_input: vsphere_resource_pool_name }
+        auto_placement: { get_input: vsphere_auto_placement }
+
+  manager_host:
+    type: cloudify.vsphere.nodes.Server
+    properties:
+      server:
+        name: { get_input: server_name }
+        template: { get_input: template }
+        cpus: { get_input: manager_server_cpus }
+        memory: { get_input: manager_server_memory }
+      install_agent: false
+      networking:
+        domain: { get_input: manager_server_domain }
+        dns_servers: ['8.8.8.8']
```

Comment:
This will not work `offline`. Don't we want to allow the image to provide its default DNS?